### PR TITLE
Theme stuff: fonts, colors, `<Text />` component

### PR DESCRIPTION
**Pull request by jessepinho**
## In this PR
- Put Penumbra fonts in the `assets/fonts` directory
- Use these fonts in both Storybook and the app itself
- Set up all the theme text variants like in Penumbra web
- Create a Storybook story for the built-in `<Text />` component, that demonstrates how to use it
- Build out theme colors in the way that Dripsy understands them — i.e., not using sub-keys

### CODE REVIEW

File: react-native-expo/.storybook/main.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ const config: StorybookConfig = {
       resolve: {
         alias: {
           // Match the tsconfig.json alias
-          '@': path.resolve(__dirname, '../src'),
+          '@': path.resolve(__dirname, '..'),
```
Comment (jessepinho) [inline react-native-expo/.storybook/main.ts:31]: This was a mistake from copying it over from elsewhere, so I fixed it.

File: react-native-expo/tsconfig.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
   "extends": "expo/tsconfig.base",
   "compilerOptions": {
     "strict": true,
+    "jsx": "react-jsx",
```
Comment (jessepinho) [inline react-native-expo/tsconfig.json:5]: Otherwise, React fragments (`<>foo</>`) don't work

File: react-native-expo/components/Text.stories.tsx
Comment (jessepinho) [inline react-native-expo/components/Text.stories.tsx:1]: image

File: react-native-expo/utils/dripsyTheme.ts [hunk 3]
```diff
@@ -4,9 +4,246 @@ import { theme } from './theme';
...
+      default: 'IosevkaTerm',
+      bold: 'IosevkaTermMedium',
+    },
+  },
+
+  fonts: {
+    root: 'Poppins',
+    mono: 'IosevkaTerm',
+    heading: 'WorkSans',
+  },
```
Comment (TalDerei) [inline react-native-expo/utils/dripsyTheme.ts:85]: assuming we've opted to reference the existing the minifront V2 component library going forward? These typography styles seem to match the fonts used in the Figma designs.

**Review by TalDerei — APPROVED** [review]
left some minor comments 👍

File: react-native-expo/components/RootLayout/FontProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { ReactNode } from 'react';
+import { useFonts } from 'expo-font';
+import dripsyTheme from '@/utils/dripsyTheme';
+
+interface FontsProps {
+  children: ReactNode;
+}
+
+export default function FontProvider({ children }: FontsProps) {
```
[the remark this answers is not in the document]
Comment (jessepinho) [reply react-native-expo/components/RootLayout/FontProvider.tsx:9]: Correct — but note that it's loading the fonts from the local file system. I imagine the load time is negligible. That said, I'll be dealing with the loading screen eventually as well and will evaluate load times then and whether this needs to be fixed.

File: react-native-expo/utils/dripsyTheme.ts (hunk 3 above)
Comment (jessepinho) [reply react-native-expo/utils/dripsyTheme.ts]: Not exactly the _component_ library, but rather the theme tokens (fonts, sizes, colors, etc.)

File: react-native-expo/components/RootLayout/FontProvider.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { ReactNode } from 'react';
+import { useFonts } from 'expo-font';
```
[the remark this answers is not in the document]
Comment (jessepinho) [reply react-native-expo/components/RootLayout/FontProvider.tsx:2]: That's just for the fonts though — not for all other styles/etc. What's your concern about styling?

[the remark this answers is not in the document]
Comment (TalDerei) [reply react-native-expo/components/RootLayout/FontProvider.tsx:2]: this was more of a general comment than a specific concern, I should have clarified; will start using `conventional comments` to make this clearer!

**Pull request merged**